app/src/test/report/fetch/ReportingFetchHelper.ts

```diff
@@ -0,0 +1,162 @@
+import { FetchTestConfig, FetchHelperConfig} from "../../shared/fetch/helpers";
+import { expect } from "chai";
+import {expectNoActions, getActions} from "../../actionHelpers";
+import {mockResult, promiseJSON} from "../../mocks/mockRemote";
+import {Sandbox} from "../../Sandbox";
+import {ErrorInfo, Result } from "../../../main/shared/models/Generated";
+import { Notification } from "../../../main/shared/actions/NotificationActions";
+import {ReportingFetcher} from "../../../main/report/sources/ReportingFetcher";
+import fetcher, { FetchOptions } from "../../../main/shared/sources/Fetcher";
+
+function mockFetcherResponse<T>(data?: Result, errorMessage?: string) {
+    data = data || {
+            status: "success",
+            data: {},
+            errors: []
+        };
+
+    const promise = new Promise<Response>(function (resolve, reject) {
+        if (errorMessage) {
+            reject(errorMessage);
+        } else {
+            resolve(promiseJSON(data));
+        }
+    });
+
+    mockReportingFetcher(promise);
+}
+
+
+function mockReportingFetcher(promise: Promise<Response>) {
+    fetcher.fetcher = new ReportingFetcher();
+    fetcher.fetcher.fetchFromReportingApi = function(urlFragment: string, options?: FetchOptions, includeToken: boolean = true) {
+        return promise;
+    };
+}
+
+export class ReportingFetchHelper<TPayload> {
```

Comment:
Again, a lot of duplication that it would be better to avoid. Maybe you could avoid the duplication if the ReportFetcher had the same interface as Fetcher. i.e. `fetch`, rather than `fetchFromReportingApi`?